### PR TITLE
Better widget highlight when resizing [MAILPOET-1819]

### PR DESCRIPTION
**Pull request by costasovo**

### CODE REVIEW

**Comment by costasovo** [comment]
@amine-mp I've added one more [commit](https://github.com/mailpoet/mailpoet/pull/1850/commits/3780828f4734aa764bf7c93c3b80e4ff98e4eaf9) with fix for flickering black borders which we caused by unwanted transition.
https://d.pr/v/LNmWBs

**Comment by costasovo** [comment]
@amine-mp I've added one more commit with a fix for broken behaviour that click on alc block doesn't open settings.

**Comment by costasovo** [comment]
@amine-mp I had to add 3 more commits to fix highlight issues for parent containers during block resizing.
https://github.com/mailpoet/mailpoet/pull/1850/commits/2ca3b15ed9176e99b8a1fff1c9bc391f4416e9c3, https://github.com/mailpoet/mailpoet/pull/1850/commits/4a900d8c38f3ed7741b9066f54521808f102a5a3, https://github.com/mailpoet/mailpoet/pull/1850/commits/4cdc20977559d6264f640cb3dcadcbf78e6d750b

File: assets/js/src/newsletter_editor/utils.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const isEventInsideElement = (event, $el) => {
```
Comment (amine-mp) [inline assets/js/src/newsletter_editor/utils.js]: You can directly do `export const ...` without the need of the export statement below.

File: assets/js/src/newsletter_editor/behaviors/HighlightEditingBehavior.js [hunk 2]
```diff
@@ -47,9 +49,25 @@ BL.HighlightEditingBehavior = Marionette.Behavior.extend({
...
+  onStopResizing: function onStopResizing(event) {
+    this.onStopEditing();
+    this.view.triggerMethod('resizeStop', event);
+  },
   onDomRefresh: function onDomRefresh() {
     if (this.isBeingEdited) {
       this.view.addHighlight();
     }
   },
+  onChildviewResizeStart: function onChildviewResizeStart() {
```
Comment (amine-mp) [inline assets/js/src/newsletter_editor/behaviors/HighlightEditingBehavior.js]: I would write onChild**V**iewResizeStart to respect the camel case rule.

Comment (costasovo) [reply assets/js/src/newsletter_editor/behaviors/HighlightEditingBehavior.js]: I would too. But it isn't an option :( This naming is required by framework: https://marionettejs.com/docs/master/events.html#event-bubbling

**Comment by costasovo** [comment]
@amine-mp I've changed the export.

**Pull request merged**